platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/net/ConnectivityReceiver.java

```diff
@@ -0,0 +1,74 @@
+package com.mapbox.mapboxsdk.net;
+
+import android.content.BroadcastReceiver;
+import android.content.Context;
+import android.content.Intent;
+import android.content.IntentFilter;
+import android.net.ConnectivityManager;
+import android.net.NetworkInfo;
+import android.util.Log;
+
+import java.lang.ref.WeakReference;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.CopyOnWriteArrayList;
+
+/**
+ * Listens for Connectivity changes
+ */
+public class ConnectivityReceiver extends BroadcastReceiver {
```

Comment:
Would it be possible to use an local broadcast receiver instead?

https://developer.android.com/reference/android/support/v4/content/LocalBroadcastManager.html